ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -560,7 +552,13 @@ export default class ConfirmTransactionBase extends Component {
           tokenSymbol={tokenSymbol}
           isUsingPaymaster={isUsingPaymaster}
         />
-        {simulationDetails}
+        {simulationData && (
```

Comment:
Just to play devil's advocate, could we have the `SimulationDetails` component own this decision and update the logic generating the event to skip if there was no data?

Reply:
The problem here is exactly that Matthew, I definitely agree what you are saying, `SimulationDetails` should decide on it's own if it will load or not. 

But we cannot early return in the very beginning of the `SimulationDetails` because of the `useSimulationMetrics` - violation of early return before hook. And if we let `useSimulationMetrics` to let run once with undefined `simulationData` it causes to race condition and letting event trigger.

Reply:
This is what happens in `develop`. Two of those `simulationData` is undefined, I forgot to show in video

https://github.com/user-attachments/assets/1f2eb0dd-f1c4-45da-aa50-b47c11d4c436

Reply:
But in that case, could we not change the logic within `useSimulationMetrics` to also early return if there is no `simulationData`?

Reply:
Guess I was a bit overthinking on the solution, you're right, it fixes the issue.